src/Init/Lean/Elab/Command.lean

```diff
@@ -518,10 +523,11 @@ levelNames      ←
       savedLevelNames
   };
 let ref := declId;
+let (id, scps) := extractMacroScopes id;
```

Comment:
This is nice. Please, document/justify why we need to extract and then restore macro scopes.